### PR TITLE
Origin/feature/updated start time

### PR DESCRIPTION
**Pull request by Vita1287**
Changed attributes for Word to include start times for the input video and the output video. Note that Patrick's changes for preProcess have been used in this PR. The original startTime has been removed so front-end may need to change.

Changes:
- In sharedTypes, Word interface was updated to include inputStartTime and outputStartTime and startTime is removed
- preProcess was refactored to become compatible with the new Word interface

### CODE REVIEW

**Review by LiamTodd — APPROVED** [review]
Looking good, chad

File: src/main/editDelete/preProcess.ts [hunk 1]
```diff
@@ -1,52 +1,74 @@
...
-  processedTranscript.words.push(lastWord);
+/**
+ * Injects extra attributes into a PartialWord to make it a full Word -
+ * @param word the word to fill attributes for
+ * @param i index of the word in the transcript
+ * @returns full Word object
+ */
+const injectAttributes: MapCallback<PartialWord, Word> = (word, i) => ({
+  ...word,
+  outputStartTime: 0,
```
Comment (rileykeane) [inline src/main/editDelete/preProcess.ts]: Can you please adjust the initial `outputStartTime` to be equal to `startTime`

File: src/sharedTypes.ts [hunk 2]
```diff
@@ -28,8 +28,9 @@ export interface Transcription {
 
 export interface Word {
   word: string;
-  startTime: number;
+  inputStartTime: number;
```
Comment (rileykeane) [inline src/sharedTypes.ts]: Have you confirmed changing this won't break anything with export?

File: src/main/editDelete/postProcess.ts [hunk 3]
```diff
@@ -1,70 +1,70 @@
-import { Transcription, Clip } from 'sharedTypes';
+// import { Transcription, Clip } from 'sharedTypes';
```
Comment (ghost) [inline src/main/editDelete/postProcess.ts]: Just wanted to clarify what is going on in this file. Is the code commented out for reason? Is it going to make a comeback?

File: src/main/editDelete/__tests__/preProcess.test.ts [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+          start_time: 11.2,
+          duration: 0.9,
+        },
+      ],
+    };
+    const duration = 15.77;
+
+    const outputTranscript = preProcessTranscript(jsonTranscript, duration);
+
+    expect(outputTranscript).toEqual({
```
Comment (rileykeane) [inline src/main/editDelete/__tests__/preProcess.test.ts:72]: I'm surprised these test cases passed. I believe they need to be updated to included the new keys in `Word`

File: src/main/editDelete/postProcess.ts (hunk 3 above)
Comment (LiamTodd) [reply src/main/editDelete/postProcess.ts]: We commented it out for now as we are going to refactor it and is not being used. Currently making a comeback lol

**Review by rileykeane — CHANGES_REQUESTED** [review]
Thanks for updating the `Word` model. 

Please just fix the initial `outputStartTime` value and also confirm that changing the `startTime` key won't break anything - it won't effect the project page in the front end but might cause issues else where. 

Also, please remove any commented out code that you don't need anymore. 

@hhen0005 can you please confirm the `outputStartTime` is what you require for video playback?

File: src/main/editDelete/__tests__/preProcess.test.ts (hunk 4 above)
Comment (rileykeane) [reply src/main/editDelete/__tests__/preProcess.test.ts:72]: I'm just now realizing you didn't create these and its just from Patrick's refactor that hasn't been merged with develop.

**Comment by Vita1287** [comment]
> Thanks for updating the `Word` model.
> 
> Please just fix the initial `outputStartTime` value and also confirm that changing the `startTime` key won't break anything - it won't effect the project page in the front end but might cause issues else where.
> 
> Also, please remove any commented out code that you don't need anymore.
> 
> @hhen0005 can you please confirm the `outputStartTime` is what you require for video playback?

**Comment by Vita1287** [comment]
Changes have been made to the following:
inputStartTime has been changed to startTime
outputStartTime is now initially assigned startTime

File: src/sharedTypes.ts (hunk 2 above)
Comment (zwan0202) [reply src/sharedTypes.ts]: We will fix the export after this PR is merged

**Review by zwan0202 — APPROVED** [review]

**Review by rileykeane — APPROVED** [review]

**Pull request merged**